packages/features/eventtypes/lib/getPublicEvent.ts

```diff
@@ -423,6 +426,18 @@ export const getPublicEvent = async (
       length: eventWithUserProfiles.length,
     });
   }
+  const membership = await prisma.membership.findFirst({
+    where: {
+      userId: currentUserId ?? -1,
+      teamId: event.teamId ?? -1,
+      accepted: true,
+      role: "MEMBER",
+    },
+  });
+
+  if (event.team?.isPrivate) {
+    users = !currentUserId || membership ? [] : users;
+  }
```

Comment:
This won't work when logged in user is owner of org but not admin of that team. He should also be able to see all the users

Reply:
also we should simplify this 

const isOwnerAdminOfTeam = await prisma.membership.findFirst({
      where: {
        userId: sessionUserId,
        teamId,
        accepted: true,
        role: {
          in: ["OWNER", "ADMIN"],
        },
      },
    });